saleor/plugins/user_email/plugin.py

```diff
@@ -410,29 +417,99 @@ def __init__(self, *args, **kwargs):
         if event not in UserNotifyEvent.CHOICES:
             return previous_value
+
         if event not in event_map:
             logger.warning(f"Missing handler for event {event}")
             return previous_value
+
         template_map = get_user_template_map(self.templates)
         if not template_map.get(event):
             return previous_value
-        event_map[event](
-            payload,
-            asdict(self.config),  # type: ignore
-            self.configuration,
-        )
+
+        event_func = event_map[event]
+        config = asdict(self.config)  # type: ignore
+        event_func(payload, config, self)
 
     @classmethod
-    def validate_plugin_configuration(cls, plugin_configuration: "PluginConfiguration"):
+    def validate_plugin_configuration(
+        cls, plugin_configuration: "PluginConfiguration", **kwargs
+    ):
         """Validate if provided configuration is correct."""
         configuration = plugin_configuration.configuration
         configuration = {item["name"]: item["value"] for item in configuration}
 
         validate_default_email_configuration(plugin_configuration, configuration)
-        validate_format_of_provided_templates(plugin_configuration, TEMPLATE_FIELDS)
+        email_templates_data = kwargs.get("email_templates_data", [])
+        validate_format_of_provided_templates(
+            plugin_configuration, email_templates_data
+        )
+
+    @classmethod
+    def save_plugin_configuration(
+        cls, plugin_configuration: "PluginConfiguration", cleaned_data
+    ):
+        current_config = plugin_configuration.configuration
+
+        configuration_to_update = []
+        email_templates_data = []
+
+        for data in cleaned_data.get("configuration", []):
+            if data["name"] in TEMPLATE_FIELDS:
+                email_templates_data.append(data)
+            else:
+                configuration_to_update.append(data)
+
+        if configuration_to_update:
+            cls._update_config_items(configuration_to_update, current_config)
+
+        if "active" in cleaned_data:
+            plugin_configuration.active = cleaned_data["active"]
+
+        cls.validate_plugin_configuration(
+            plugin_configuration, email_templates_data=email_templates_data
+        )
+        cls.pre_save_plugin_configuration(plugin_configuration)
+
+        plugin_configuration.save()
+
+        for et_data in email_templates_data:
+            if et_data["value"] != DEFAULT_EMAIL_VALUE:
+                EmailTemplate.objects.update_or_create(
+                    name=et_data["name"],
+                    plugin_configuration=plugin_configuration,
+                    defaults={"value": et_data["value"]},
+                )
+
+        if plugin_configuration.configuration:
+            # Let's add a translated descriptions and labels
+            cls._append_config_structure(plugin_configuration.configuration)
+        return plugin_configuration
```

Comment:
It's not the same as in the `saleor/plugins/admin_email/plugin.py` `save_plugin_configuration` method? Maybe we can create utils from this?

Reply:
Thanks for catching this anyway :)